app/scripts/directives/nav.js

```diff
@@ -342,11 +355,98 @@ angular.module('openshiftConsole')
               $scope.currentProject = projects[$scope.currentProjectName];
               updateOptions();
             });
+
+            var includeTemplates = !CatalogService.isTemplateServiceBrokerEnabled();
+            Catalog.getCatalogItems(includeTemplates).then(_.spread(function(items, errorMessage) {
```

Comment:
This is potentially a lot of data, so let's be careful about loading it when not needed.

*  We shouldn't reload all of this data when switching projects since it's cluster scoped and doesn't change often. It looks like we request it again on project changes.
* We should only load the catalog items when the user has authority to add to a project.
* I suggest we load the data once for the console and cache it in a service. Then it can be used for the catalog and nav search and anywhere else we need it.

Reply:
Should we do the same for service plans? We would have to watch those though as they can change and get filtered by the 'removedFromBrokerCatalog' field.

Reply:
We allow search when the user is not allowed to add. This gives the ability to at least see what catalog items are out there and get their info.

We can remove search in project context if we think this is not necessary.

Reply:
> Should we do the same for service plans?

Yes. Plans contain the JSON schema describing the parameters, so the response could get large if you have hundreds or thousands. QE has seen the request take up to 10s in their environment.

https://bugzilla.redhat.com/show_bug.cgi?id=1512858

> We would have to watch those though as they can change and get filtered by the 'removedFromBrokerCatalog' field.

I don't think it's worth watching since this happens infrequently. (I'm not sure offhand if watch is supported for plans.)

It might make sense to lazy load plans for just the service class when opening the dialog instead of loading all of them up front. (Not necessarily for this PR, though.)

Reply:
> We allow search when the user is not allowed to add. This gives the ability to at least see what catalog items are out there and get their info.

That seems a little odd to me. We do have the landing page if the user wants to browse / search.

Reply:
Added `CatalogService.getCatalogItems` which caches the catalog items

>It might make sense to lazy load plans for just the service class when opening the dialog instead of loading all of them up front. (Not necessarily for this PR, though.)

Updated to only get the plans for the class when opening the dialog

>That seems a little odd to me. We do have the landing page if the user wants to browse / search.

Added the check